fcrepo4/fcrepo4
# Implement containment triples service

and wire index into Create/Delete services.

**JIRA Ticket**: https://jira.lyrasis.org/browse/FCREPO-3164

# What does this Pull Request do?
Creates a Containment Triples service to get the containment.
Wires the containment index into the CreateResourceService and DeleteResourceService to add/remove relationships.
Alters the ContainmentIndex API to add some necessary methods and moves some from using `Transaction` and `FedoraResource` to using their `String` ids.

# How should this be tested?

1. Start it up and add a new resource
1. Check root to see the ldp:contains triple
1. Try adding children using compound Slugs (ie. `Slug: A` then `Slug: A/B`)
1. Try deleting children and see that the containment triple disappears from the parent.

# Interested parties
@fcrepo4/committers @pwinckles
